waku/README.md

```diff
@@ -0,0 +1,301 @@
+# Waku Whisper Specification
+
+> Version 0.0.5 (Draft)
+>
+> Authors: Oskar Thorén oskar@status.im, Dean Eigenmann dean@status.im
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Specification](#specification)
+    - [Use of RLPx transport protocol](#use-of-rlpx-transport-protocol)
+    - [Packet Codes](#packet-codes)
+    - [Packet Format and Usage](#packet-format-and-usage)
+    - [Whisper Envelope](#whisper-envelope)
+    - [Contents of Data Field of the Message (Optional)](#contents-of-data-field-of-the-message-optional)
+    - [Payload Encryption](#payload-encryption)
+- [Light node](#light-node)
+    - [Rationale](#rationale)
+    - [Considerations](#considerations)
+- [Rationale](#rationale-1)
+- [Backwards Compatibility](#backwards-compatibility)
+    - [WakuWhisper bridging](#wakuwhisper-bridging)
+- [Forwards Compatibility](#forwards-compatibility)
+- [Security Considerations](#security-considerations)
+- [Implementation](#implementation)
+- [Changelog](#changelog)
+    - [Differences between shh/6 waku/0](#differences-between-shh6-waku0-1)
+- [Copyright](#copyright)
+
+## Abstract
+
+This specification describes the format of Waku messages within the ÐΞVp2p Wire Protocol. This spec substitutes [EIP- 627](https://eips.ethereum.org/EIPS/eip-627). Waku is a fork of the original Whisper protocol that enables better scalability and offline messaging, at the cost of some metadata protection guarantees. It does this through (a) light node support (b) mailserver (c) waku mode and (d) basic rate limiting.
```

Comment:
I would change `mailserver` to something like `historic messages`. `mailserver` is confusing and I don't think we should continue using this terminology.

Reply:
`waku mode` isn't all these things all together? It is not mentioned even once in the spec later.

Reply:
@adambabik I don't think we should move away from mailserver as that is the naming most people are familiar with also it works as an ugly thing that we are motivated to move away from. Once we move to something like remotelog I am all for calling it historic messages.

Reply:
I rewrote this to reflect current state:

"[...]Waku is a fork of the original Whisper protocol that enables better usability for resource restricted devices, such as mostly-offline bandwidth-constrained smartphones. It does this primarily through (a) light node support and (b) historic messages (with a mailserver). Additionally, other experimental features for better scalability and DDoS resistance are under development and will be part of future versions."

we can add back rate limiting and explain waku mode for v1 (left in todo comment).